### ISSUE TITLE
Feeling dumb...

### ISSUE THREAD

**Issue by teshiburu2020** [bexem/PlexCache#13]
I'm currently going through the setup.py on this in PowerShell on server 2019, when it gets to the point of asking about my cache drive it defaults to /mnt/cache which is the correct location on my unraid server... but it's not giving me an option to specify any unraid details? have i missed something? or can this not communicate with my unraid server, and will i need to "cache" it on the windows machine locally?

**Comment by bexem** [comment]
Hi there! 
The setup script is only to format nicely the settings file, the onlyexternal communication is with the plex server. 
The cache script need to be run directly on the plexserver machine or at least in a machine who as access to the same paths and able to move files around.

**Comment by teshiburu2020** [comment]
Then I'll need to be running this from the unraid server directly then as my windows VM for Plex can't see mnt cache....

**Comment by bexem** [comment]
I suppose so, I personally run it directly on the unraid server (which is also my plex server), I think it makes sense as it has direct access to the directories avoiding any network/protocol overhead when moving files around.

**Comment by teshiburu2020** [comment]
ok, so i moved it to unraid, and i think my settings is fine: 
```{
    "PLEX_URL": "http://192.168.1.220:32400",
    "PLEX_TOKEN": "SUPER SECRET",
    "plex_source": "/",
    "plex_library_folders": [
        "P:\\Movies",
        "P:\\TV Shows"
    ],
    "valid_sections": [
        1,
        2
    ],
    "number_episodes": 5,
    "users_toggle": true,
    "skip_users": [
        null,
        null,
        null,
        null,
        null,
        null,
        null
    ],
    "watchlist_toggle": true,
    "watchlist_episodes": 1,
    "watchlist_cache_expiry": 6,
    "watched_cache_expiry": 48,
    "days_to_monitor": 99,
    "cache_dir": "/mnt/cache/",
    "real_source": "/mnt/user/",
    "nas_library_folders": [
        "mnt/MediaShare/Movies",
        "/mnt/MediaShare/TV Shows/"
    ],
    "unraid": true,
    "watched_move": false,
    "skip": true,
    "max_concurrent_moves_cache": 5,
    "max_concurrent_moves_array": 2,
    "debug": false,
    "firststart": false
}```



```2023-05-19 09:32:00,338 - INFO - Total size of media files to be moved to cache: 0.00 KB
2023-05-19 09:32:00,338 - INFO - Nothing to move to cache
2023-05-19 09:32:00,338 - INFO - Script executed```

**Comment by bexem** [comment]
`{
    "PLEX_URL": "https://plex.bexem.xyz",
    "PLEX_TOKEN": "********",
    "plex_source": "/media/",
    "plex_library_folders": [
        "movies",
        "anime",
        "tvseries"
    ],
    "valid_sections": [
        1,
        4,
        2
    ],
    "number_episodes": 10,
    "users_toggle": true,
    "skip_users": [],
    "watchlist_toggle": true,
    "watchlist_episodes": 5,
    "watchlist_cache_expiry": 48,
    "days_to_monitor": 183,
    "watched_move": true,
    "watched_cache_expiry": 12,
    "cache_dir": "/mnt/cache/",
    "real_source": "/mnt/user/",
    "nas_library_folders": [
        "movies",
        "anime",
        "tvseries"
    ],
    "unraid": true,
    "max_concurrent_moves_array": 2,
    "max_concurrent_moves_cache": 5,
    "skip": false,
    "debug": false,
    "firststart": false
}`

Your plex library folders and the nas library folder appear to be wrong. Have a look at my settings and compare.

**Comment by teshiburu2020** [comment]
So is your media in this example located at 

/Mnt/user/movies
/mnt/user/tvseries

? 

Which would suggest I need to change my real source to /mnt/user/mediashare

And then library folders to 
Movies
TV Shows

?

**Comment by bexem** [comment]
Yes, 
my movies are in /mnt/user/movies
my series are in /mnt/user/tvseries
my anime are in /mnt/user/anime

As you can see the real source is /mnt/user/
and the nas_library_folders are the name of the folder above (movies, tvseries and anime)

Also your plex source was wrong at least judging from here, as it has so be the folder the container access the media from, with the plex_library_folders set similarly to the nas folder but according to how you have it configured (mine are exactly the same but you could have movies in the nas folder corresponding to films in the plex library as an example).

**Comment by bexem** [comment]
I agree the script should handle that, unfortunately I don’t really have a way to test but I will try my best to fix it anyway.
I will probably have a look into it this afternoon. (Night shift, GTM Tz)

Just so it’s clear for me later on: you are running plex on windows directly and not via docker container, correct?
Also, where would run the script? UnRaid? 
If you don’t mind re-sharing your updated setting file?

**Comment by teshiburu2020** [comment]
I'm running Plex directly on Windows yes. 

Unraid is running the script

I think I found a solution which I'll share maybe point you the right way?

**Comment by bexem** [comment]
Sure, anything can help 😊

**Comment by teshiburu2020** [comment]
On line 314 I made this change it's crude but seems to have worked 

```file_path = file_path.replace(folder, nas_library_folders[j]).replace("\\", "/") # Replace the plex library folder with the corresponding NAS library folder ```

**Comment by bexem** [comment]
> On line 314 I made this change it's crude but seems to have worked
> 
> `file_path = file_path.replace(folder, nas_library_folders[j]).replace("\\", "/") # Replace the plex library folder with the corresponding NAS library folder `

That is actually a not a so dirty solution but I will implement it soon in a conditional manner.
Could be unrelated but I rather be sure of it, would you mind telling me how do you have set up the script_folder variable in the cache script? Because you are running it on windows I guess you have a windows formatted folder syntax set?

**Comment by teshiburu2020** [comment]
I'm running it on unraid

**Comment by bexem** [comment]
Thank you, I'm working on a universal solution and your case seems perfect for this. So, whenever you can, please attach a redacted settings file you are currently using because I want to be sure the script outputs correctly once I've implemented the fix.

**Comment by bexem** [comment]
Just an update: I've uploaded a setup_test.py script, it will produce a settings_test.json file (to avoid introducing error in a working file). 
I'm leaning towards having the setup script doing the checks and set the paths correctly rather than introducing more logic in the cache script. I might still change my mind, but for testing, the setup script is faster.
If you don't mind testing it and posting the result. 

Thank you for your time anyway ☺️

**Comment by teshiburu2020** [comment]
One issue ive noticed in setup.py in both the test and existing is that "watched_cache_expiry" doesn't get generated :)

**Comment by bexem** [comment]
Well spotted, that was a minor bug that shows up only if answered no to moving the watched media! Fixed! 
Any other issues?

**Comment by teshiburu2020** [comment]
OK, so the "settings_test" or something funky is going on. 

```{
    "PLEX_URL": "http://192.168.1.220:32400",
    "PLEX_TOKEN": "XXXXXXXX",
    "plex_source": "P:/",
    "plex_library_folders": [
        "P:\\Movies",
        "P:\\TV Shows"
    ],
    "valid_sections": [
        1,
        2
    ],
    "number_episodes": 5,
    "users_toggle": false,
    "skip_users":[],
    "watchlist_toggle": true,
    "watchlist_episodes": 1,
    "watchlist_cache_expiry": 24,
    "watched_cache_expiry": 24,
    "days_to_monitor": 99,
    "cache_dir": "/mnt/cache",
    "real_source": "/mnt/user/MediaShare/",
    "nas_library_folders": [
        "Movies",
        "TV Shows"
    ],
    "unraid": true,
    "watched_move": true,
    "skip": true,
    "max_concurrent_moves_cache": 5,
    "max_concurrent_moves_array": 2,
    "debug": false,
    "firststart": false
}
```


This is my settings file, it seems to ALWAYS set plex_source as p:/Movies and not just P:/ im assuming a bug here? as it just seems to always put the folder for the "first" library i select here so if i skipped movies and said yes to TV shows it would dbe P:/TV Shows ive been changing this to P:/ 

But the issue now seems to be that it thinks nothing is to be moved? 

```
2023-05-22 08:17:27,610 - INFO - Initializing script...
2023-05-22 08:17:27,610 - INFO - Checking paths and permissions...
2023-05-22 08:17:27,611 - INFO - Loading watchlist media from cache...
2023-05-22 08:17:27,789 - INFO - Fetching teshiburu's onDeck media...
2023-05-22 08:17:29,083 - INFO - Editing file paths...
2023-05-22 08:17:29,084 - INFO - Fetching subtitles...
2023-05-22 08:17:29,089 - INFO - Loading watched media from cache...
2023-05-22 08:17:29,089 - INFO - Filtering media files array...
2023-05-22 08:17:29,422 - INFO - Total size of media files to be moved to array: 0.00 KB
2023-05-22 08:17:29,422 - INFO - Nothing to move to array
2023-05-22 08:17:29,423 - INFO - Filtering media files cache...
2023-05-22 08:17:29,423 - INFO - Total size of media files to be moved to cache: 0.00 KB
2023-05-22 08:17:29,423 - INFO - Nothing to move to cache
2023-05-22 08:17:29,423 - INFO - Script executed
Thank you for using my script github.com/bexem/PlexCache
```

**Comment by teshiburu2020** [comment]
Edit: fixed the cache path above, but that didnt fix the issue, nothing to move apparently?

**Comment by bexem** [comment]
> Edit: fixed the cache path above, but that didnt fix the issue, nothing to move apparently?

Can you share your log file? (With debug mode set to true), it will help debugging. 

It's very likely all down to how the script(s) is managing the paths and trying to retrieve the files and checking on the filesytem their existence. I honestely hoped python (as I've read) was going to deal a bit better with path between windows and posix...but apparently not!

**Comment by teshiburu2020** [comment]
It seems that no "cache" paths are being generated? 

[Log.txt](https://github.com/bexem/PlexCache/files/11531692/Log.txt)

**Comment by bexem** [comment]
That seems to be the issue, I need to understand why. I'm looking into it...

Not related, but does the script work (well, kind off apparently) if your plex folders are ['Movies', 'TV Shows'] instead of ['P:\\Movies', 'P:\\TV Shows'] ?

**Comment by teshiburu2020** [comment]
let me try

**Comment by bexem** [comment]
But it still fetches the files correctly?

Either way:
The problem is in the fact that is running on linux but is using a paths from windows and linux, I basically need to use a different library to make the script understand what is dealing with first and then correct the paths.

I have no way to test your scenario because both the script and plex server run on linux, I've introduced a plex_cache_test script, it might not work but it's worth a shot. I've also updated the setup_test, if you don't mind retesting that one, please post the exact settings_test.json, just comment how the paths should actually be without correcting it directly. 
For the plexcache_test, the log should be enough. 

Thank you.

**Comment by teshiburu2020** [comment]
This appears to be generating extra braces? 

```
{}    "PLEX_URL": "http://192.168.1.220:32400",
    "PLEX_TOKEN": 

```

**Comment by teshiburu2020** [comment]
ALso seems if i tell it to not import other users it fails to generate this line 

```
"skip_users": [],
```

which it then moans about :)

**Comment by bexem** [comment]
Funny how is giving issues on stuff I haven't even touched 🤣 I'm checking now...

**Comment by teshiburu2020** [comment]
ok - fixed issues in settings_test.json which i will comment and upload for you :) 

but this is the log file after trying to run 

```
2023-05-22 16:03:17,393 - INFO - Editing file paths...
2023-05-22 16:03:17,393 - INFO - Fetching subtitles...
2023-05-22 16:03:17,397 - INFO - Loading watched media from cache...
2023-05-22 16:03:17,397 - INFO - Filtering media files array...
2023-05-22 16:03:17,397 - WARNING - Error: file path P:\TV Shows\A Million Little Things\Season 5\A Million Little Things.-.S05E13.-.one.big.thing.WEBDL-1080p.-.h264 EAC3.mkv is not inside real source path /mnt/user/MediaShare
2023-05-22 16:03:17,397 - INFO - File: P:\TV Shows\A Million Little Things\Season 5\A Million Little Things.-.S05E13.-.one.big.thing.WEBDL-1080p.-.h264 EAC3.mkv
2023-05-22 16:03:17,397 - INFO - Cache_path: None
2023-05-22 16:03:17,397 - INFO - Cache_file_name: None
2023-05-22 16:03:17,397 - ERROR - Error checking free space and moving media files: stat: path should be string, bytes, os.PathLike or integer, not NoneType


```

**Comment by teshiburu2020** [comment]
[settings_test_upload.txt](https://github.com/bexem/PlexCache/files/11532822/settings_test_upload.txt)

settings for you :)

**Comment by teshiburu2020** [comment]
[settings_test_upload.txt](https://github.com/bexem/PlexCache/files/11532889/settings_test_upload.txt)
settings :)

**Comment by bexem** [comment]
Thank you honestly for testing it, it's a learning experience for me. 

So I've updated (once again) the setup script, let me know it works now, should get the operating system on which plex is running automatically, fixed the brakets (honestly, such a weird one!), it should now put the right folders in the settings, and also fixed the cache variables not saving as they should have.

I'm gonna have a look at the cache script in the meantime.

**Comment by teshiburu2020** [comment]
Awesome I'm out at present I'll check the setup script when I get home. 

I may not have the knowledge to do what you are haha, but i can test and click buttons haha

**Comment by bexem** [comment]
> Awesome I'm out at present I'll check the setup script when I get home.
> 
> I may not have the knowledge to do what you are haha, but i can test and click buttons haha

Ahahah it works fine for me!

Also, I've updated the plexache test as well, I hope it works but I doubt...I'm not that lucky 🤣 
Let me know, as usual the log is the best way to debug the cache script.

**Comment by teshiburu2020** [comment]
apparently "skip_users":[], is still not being added in when its told to not import other users?

**Comment by teshiburu2020** [comment]
nor is "watched_cache_expiry":

**Comment by teshiburu2020** [comment]
```
{
    "PLEX_URL": "http://192.168.1.220:32400",
    "PLEX_TOKEN": "XXXX",
    "plex_source": "P:\\",
    "plex_library_folders": [
        "P:\\Movies\\",
        "P:\\TV Shows\\"
    ],
    "valid_sections": [
        1,
        2
    ],
    "number_episodes": 5,
    "users_toggle": false,
    "watchlist_toggle": true,
    "watchlist_episodes": 1,
    "watchlist_cache_expiry": 24,
    "days_to_monitor": 99,
    "cache_dir": "\\mnt\\cache\\MediaShare\\",
    "real_source": "\\mnt\\user\\MediaShare\\",
    "nas_library_folders": [
        "\\Movies\\",
        "\\TV Shows\\"
    ],
    "unraid": false,
    "watched_move": true,
    "skip": true,
    "max_concurrent_moves_cache": 5,
    "max_concurrent_moves_array": 2,
    "debug": true,
    "firststart": false
}
```

Generated settings_test.json

**Comment by teshiburu2020** [comment]
[plexcache_script_latest (1).log](https://github.com/bexem/PlexCache/files/11534772/plexcache_script_latest.1.log)

dropped days to monitor down to1 for you to make the log file smaller ;)

**Comment by bexem** [comment]
Oddly enough it was adding both watched_cache_expiry and skip_users when tested, but I've issued another fix, it hopefully fixes the paths as well as I am not happy about them.
I'm gonna have a look at the logs now.

**Comment by teshiburu2020** [comment]
I wish i knew what i was doing to break it between you and me haha but now when it asks me to confirm paths it doesnt tell me for which library i only know because ive done it so much now haha 

<img width="809" alt="image" src="https://github.com/bexem/PlexCache/assets/65967646/30f457ab-5cda-41b3-8609-8c13e1097775">


and the script failed here and didn't proceed further 

<img width="566" alt="image" src="https://github.com/bexem/PlexCache/assets/65967646/4ce06a37-52c2-4b4f-95d7-176121da24de">

**Comment by bexem** [comment]
I hope I get this right this time 😅
Let's try again?

**Comment by teshiburu2020** [comment]
![image](https://github.com/bexem/PlexCache/assets/65967646/7558a8ae-bdaf-43ff-8047-a7d12b0a203a)
Still getting the exact same bugs as last night. Fully removed all traces from my server and downloaded fresh from Github

**Comment by bexem** [comment]
I'm sorry about that, I guess the issue is actually in the conversion. I have not fixed that because I really need to see how it is before and after the conversion, it will print when asking about the plex library before and after, but the logic remained the same.
About the array to cache (Just spotted the writing error as well), should be finally fixed now. I must have mixed up between the two enviroments I'm testing 🤦🏻‍♂️

**Comment by teshiburu2020** [comment]
OK - so is there a fixed version of the setup_test.py? i see a new plexcache_test.py but not setup_test :)

**Comment by bexem** [comment]
You know the titles of this threat? that's me. I don't know what the heck happened with VSCode but it basically replaced the content of the plexcache_test script with the setup_test one...it's fixed now...So now the setup_test is the one I was talking about.

**Comment by teshiburu2020** [comment]
hahah - oh dear, im sorry to put you through this, have you got a Ko-Fi or a paypal i can drop you a coffee/beer haha!

**Comment by teshiburu2020** [comment]
![image](https://github.com/bexem/PlexCache/assets/65967646/f59399c2-c412-4f03-a54e-555c29aec736)

not sure if this is right? 

![image](https://github.com/bexem/PlexCache/assets/65967646/f0c52ed9-be50-4c27-bcc2-1d62da8ac84d)

also still not giving me a library to confirm the path for :) 

generate settings file is GOOOD
[settings_test.json.txt](https://github.com/bexem/PlexCache/files/11544763/settings_test.json.txt)

something is not right here though hehe - log file was MAHOOSIVE had to zip it up for you as i didn't want to remove anything 

[plexcache_script_latest.zip](https://github.com/bexem/PlexCache/files/11544796/plexcache_script_latest.zip)

![image](https://github.com/bexem/PlexCache/assets/65967646/fb521cce-82e2-436d-b9ad-62e5f5cf80da)

**Comment by bexem** [comment]
So, the cache script I'm leaving it for the time being because I want to be sure it gets the right paths first (in a specific way) so I can start working in a proper way to fix it.  But thank you anyway, I will definitely have a look as it will help me!

Regarding the the setup output: I can see more or less where the issues are now, I just need to make the right logic to format the path. 

I've updated it now, I hope it get it right...let me know 🤞🏻🙏🏻

_Unfortunately I have to do some other stuff today, but I will still try and work on  it later on but I guess we are both on different timezones. I'll keep the thread updated._

**Comment by teshiburu2020** [comment]
No worries I'll wait for you to let me know there is a new thing to test 

As for PayPal and such do let me know!

**Comment by bexem** [comment]
I’ll deserve a coffee only if I solve it 😁 But thank you 😊 

I have updated it with a new logic, might not have been obvious from my comment, apologies.

**Comment by teshiburu2020** [comment]
[plexcache_script_latest.log](https://github.com/bexem/PlexCache/files/11545402/plexcache_script_latest.log)

latest log :) 
[settings_test.json.txt](https://github.com/bexem/PlexCache/files/11545414/settings_test.json.txt)

and settings file :)

**Comment by bexem** [comment]
> Do you think the issues lay in the setup or rather that's the place to get it right for the cache script to work as is

I don't think it works as it is but I rather have a standardised settings file on which the script files can work on reliably. I know the "main" issue is in the caching logic but because the problem is due to the paths, I want to be sure I'm working always with the same style of paths rather than having a mix&match.

**Comment by bexem** [comment]
Sorry about that 😞
I've theoretically fixed it...

**Comment by bexem** [comment]
So...I've made progress on both scripts, and they seem to be functioning properly from my end. However, I haven't been able to test the Windows paths, so there's a possibility that they might not work for you.

I've streamlined and optimized some of the functions in the setup script. Although it will ask you about the operating system you're using, I don't plan on using that information in the future because the cache script now detects the operating system and dynamically converts the paths accordingly. It should ideally handle the paths correctly, even when running on Windows. Additionally, I've incorporated a Python utility for file copying, which is not specific to Linux.

I'm hopeful that these changes will work, but I'm still a bit skeptical. Nonetheless, I believe we're heading in the right direction, and it shouldn't be too long before you're up and running (plus, I have a couple of days off).

**Comment by teshiburu2020** [comment]
Ok - I think we MAY be there one potential issue with setup_test.py in that when it generated my paths for:

cache_dir - which should have been /mnt/cache/MediaShare 
real_source -  which should have been /mnt/user/MediaShare 

it generated these

\\mnt\\cache\\MediaShare
\\mnt\\user\\MediaShare 

as you can see here 
[settings_test.old.txt](https://github.com/bexem/PlexCache/files/11551646/settings_test.old.txt)

When then running plexcache_test 

The following errors occurred 

```
 2023-05-24 08:20:38,108 - ERROR - Error checking free space and moving media files: [Errno 2] No such file or directory: '\\mnt\\user\\MediaShare\\/Movies/Ant-Man and the Wasp Quantumania (2023)/Ant-Man and the Wasp Quantumania (2023) WEBRip-1080p.mkv'
```

And as youc an see in the log file all paths were rendered like this : 

[plexcache_test_script_latest.old.log](https://github.com/bexem/PlexCache/files/11551654/plexcache_test_script_latest.old.log)

whereas if i "corrected" the settings_test.json file to have the paths like this 

/mnt/cache/MediaShare 
/mnt/user/MediaShare 

this was the final settings_test file 

[settings_test.txt](https://github.com/bexem/PlexCache/files/11551667/settings_test.txt)

it seemed to yield positive results in the log file insinutating that content would be moved: 

[plexcache_test_script_latest.log](https://github.com/bexem/PlexCache/files/11551673/plexcache_test_script_latest.log)

**Comment by teshiburu2020** [comment]
on further review of the successful log file above i spotted that the cache_dir and real_dir ended up with a slash at the end of the path, and a slash on the media directory as seen here between MediaShare and TV Shows 

```
2023-05-24 08:23:18,421 - INFO - ('/mnt/user/MediaShare//TV Shows/Bitten/Season 2/Bitten.-.S02E10.-.Fine.Temporum.WEBDL-1080p.-.x264 EAC3.mkv', '/mnt/cache/MediaShare/TV Shows/Bitten/Season 2')

```

The rest of the path seems fine, so i assume tidying up this would allow it to move data :)

**Comment by bexem** [comment]
Sorry I'm a bit confused, probably my brain hasn't woken up completely yet and likely I'm just missing the obvious:

The setup script is giving you exactly what paths? 

As far I've understood it gave you:
- \mnt\cache\MediaShare
- \mnt\user\MediaShare
That sounds so weird to me because the logic asks you specifically for which OS you want to run the script and convert the path to the right syntax and should instead be:
- /mnt/cache/MediaShare/
- /mnt/user/MediaShare/

But then you noticed that the trailing "/" at the end of this path is actually giving you issue with the cache script and if removed it actually works?

Sorry if you have to ELI5 to me, just need another coffee today I guess 🤦🏻‍♂️😅

**Comment by teshiburu2020** [comment]
Hey - we all have that first thing in the morning wake up issue haha! and it seems i did as well, i absent mindedly typed "Windows"... my brain farted it seems. 

Now the settings file looks correct and plexcache seems to work? 
[settings_test.json.txt](https://github.com/bexem/PlexCache/files/11555509/settings_test.json.txt)


[plexcache_test_script_latest.log](https://github.com/bexem/PlexCache/files/11555511/plexcache_test_script_latest.log)

I'm gonna give the script a run and see what happens :)

**Comment by bexem** [comment]
So...........
I've updated the scripts now...happy to test? 
I hope it works!!!

**Comment by teshiburu2020** [comment]
I will give it a go in a short while :)

**Comment by teshiburu2020** [comment]
ok tested, and we seem to have taken a step back. 
file paths seem ok now. 
but it was missing 
```
    "watched_move": true,
    "debug": true
```

[settings_test_new.txt](https://github.com/bexem/PlexCache/files/11555799/settings_test_new.txt)

And it decided nothing needed to be moved? 

[plexcache_test_script_latest (1).log](https://github.com/bexem/PlexCache/files/11555795/plexcache_test_script_latest.1.log)

**Comment by teshiburu2020** [comment]
I am wondering if its because some of these are present on cache already? im just running a mover to test will update soon

**Comment by teshiburu2020** [comment]
Not sure which of the debug logics i got inbetween you updating and editing, but this time it seemed to work :) 

I had a positive looking log file, now to run a test and see if it moves... 

[plexcache_test_script_latest (2).log](https://github.com/bexem/PlexCache/files/11556147/plexcache_test_script_latest.2.log)

I'm assuming when it says a file has moved if i go to /mnt/cache/MediaShare - the relevant folder i should see the files present there

**Comment by bexem** [comment]
If you don't mind testing it but it should work.

The log seems fine, it should work and move all the files correctly. 

I notice the watched media not parsed? Or is it because of the caching? Let me know, I did not touch the logic and it seems to be still working fine on my side, but let me know if there is an issue on your end.

My favourite word apparently is "should" 🤣

PS: I like that Jurassic Park in there!

**Comment by bexem** [comment]
No the Jurassic Park was completely random, I noticed while scrolling up in the log file. 

>I'm assuming the section after "moving media files to the cache" is what is actually going to move?

Absolutely yes.

**Comment by bexem** [comment]
Nice! Yeah the command is correct, path look fine and as you've said the space is actually decreasing..
I shall wait for your confirmation to call it a win!

**Comment by teshiburu2020** [comment]
Second run says nothing to move to cache, and nothing to move to array. 

![image](https://github.com/bexem/PlexCache/assets/65967646/b9db18c8-0f21-4707-874f-a1d6067ef7c4)

I suppose with this - the only "issue" is to determine when to run this alongside unraids mover?

**Comment by teshiburu2020** [comment]
So you move all from cache when you get above 80% usage on your cache drive?

**Comment by teshiburu2020** [comment]
Yeah  - ive got backblaze running on that so not an issue there for backups i dont think. 

ok - Chronos i couldnt get my head around not to mention having just done all the scripts in my system drive haha!

**Comment by bexem** [comment]
> Yeah - ive got backblaze running on that so not an issue there for backups i dont think.

Yeah that should be more than enough!

> ok - Chronos i couldnt get my head around not to mention having just done all the scripts in my system drive haha!
I can post the screenshots of my chronos configuration if it is of any help?

**Comment by teshiburu2020** [comment]
> > Yeah - ive got backblaze running on that so not an issue there for backups i dont think.
> 
> Yeah that should be more than enough!
> 
> > ok - Chronos i couldnt get my head around not to mention having just done all the scripts in my system drive haha!
> 
> I can post the screenshots of my chronos configuration if it is of any help?

Oh yeah that would really help!

**Comment by bexem** [comment]
![Screenshot 2023-05-24 at 16-45-52 Chronos](https://github.com/bexem/PlexCache/assets/13065327/4a00c941-ca9f-48bf-b3af-b51541dfb6ad)
![Screenshot 2023-05-24 at 16-45-33 Chronos](https://github.com/bexem/PlexCache/assets/13065327/b354c689-05bb-4282-8114-1716ebc0ecf8)
![Screenshot 2023-05-24 at 16-44-10 Tower_UpdateContainer](https://github.com/bexem/PlexCache/assets/13065327/03f49e55-3ed6-47c0-9633-7233925e6775)


Use [chron.guru](https://crontab.guru/#5_7_*_*_*) For the chron syntax

**Comment by teshiburu2020** [comment]
![image](https://github.com/bexem/PlexCache/assets/65967646/ab3605bd-efa3-47c3-9668-148d90142cb9)

Looks good to me - you Mr Bexem have earnt many cups of coffee :)

**Comment by bexem** [comment]
> Spot on no worries, ill reach out with issues dont worry haha!

> When you do update the cache script i just paste it back into Chronos right :)

Hey, thank you again for the support! I just wanted to ask a small favour:
In case you have a couple of minutes, would you mind testing the new _rc scripts? 
I just want to be sure they work for you as well before merging. 
I've left the old _test for the cache script as just in case because I know it works for both of us.

**Comment by teshiburu2020** [comment]
Sure thing ill take a look for you in a bit, been one of those days here so far!

**Comment by teshiburu2020** [comment]
Managed to test all of these _rc releases and they work perfectly

**Comment by bexem** [comment]
> Managed to test all of these _rc releases and they work perfectly

Amazing! Thank you so much for testing them out!!!

They are now completely merged in the main branch (I was ready to backtrack!).

**Comment by teshiburu2020** [comment]
No worries sorry for the delay in letting you know, I wanted to let it run a few days :)
